doc/source/admin/container_resolvers.rst

```diff
@@ -0,0 +1,276 @@
+.. _container_resolvers:
+
+Container Resolvers in Galaxy
+=============================
+
+A container resolver tries to get a container description, i.e. the information
+(URI/path to the container image, ...) that is needed to execute a tool, given the
+requirements specified in this tool. Galaxy implements various container resolvers
+that are suitable for different needs. 
+
+Galaxy tries to execute jobs using containers if they are send
+to execution environments (previously called destinations) with either 
+
+- `docker_enabled <https://github.com/galaxyproject/galaxy/blob/0742d6e27702c60d1b8fe358ae03a267e3f252c3/lib/galaxy/config/sample/job_conf.sample.yml#L419>`_ or
+- `singularity_enabled <https://github.com/galaxyproject/galaxy/blob/0742d6e27702c60d1b8fe358ae03a267e3f252c3/lib/galaxy/config/sample/job_conf.sample.yml#L556>`_
+
+enabled. Note the while the links above examplify this for local execution
+environments this works for any environment as long as ``docker`` or ``singularity``
+are available.
+
+For jobs that are sent to such an execution environment Galaxy tries
+to obtain a container description by sequentially executing the container
+resolvers that are configured for the Galaxy instance (see below).
+The job is then executed using the description returned by the first successful
+container resolver.
+If all configured container resolvers failed, i.e. no container description
+could be obtained, the tool is by default excuted using 
+:doc:`standard dependency resolvers <dependency_resolvers>`, e.g. ``conda``.
+Alternatively, if the excution environment specifies
+`require_container <https://github.com/galaxyproject/galaxy/blob/0742d6e27702c60d1b8fe358ae03a267e3f252c3/lib/galaxy/config/sample/job_conf.sample.yml#L528>`_
+the job fails in this case (TODO to be tested? or is it already).
+
+Configuration:
+--------------
+
+The list of container resolvers is defined using YAML that can be
+specified in an extra file (``container_resolvers_config_file``) or
+inline the Galaxy configuration (``container_resolvers``). 
+During the container resolution the resolvers specified in this
+list are sequentially applied stopping at the first resolver that
+yields a container description. 
+
+The YAML defines a list of container resolvers and their properties.
+This might look as follows:
+
+.. code-block:: yaml
+
+   - type: "cached_explicit_singularity"
+   - type: "cached_mulled_singularity"
+   - type: "mulled_singularity"
+     auto_install: False
+
+would define a list of three container resolvers, and the last setting an option
```

Comment:
This might be a good opportunity as well to add a sample file in YAML for this config file, that reflects how the same content of the XML would be written (sometimes mapping XML attributes and children to YAML is not always straightforward, specially if one lacks an understanding of the code loading this config). This could help in the transition from XML to YAML files for people.

Reply:
Indeed. 

And maybe also a schema definition? But I don't understand those yet. Guess it wont be more than a mapping anyway (except maybe for the mapping resolver).

Reply:
+1 to YAML. Also, the container config is generally so simple that I'm putting it inline in galaxy.yml anyway.